### DISCUSSION TITLE
issue with reading rgba image

### DISCUSSION THREAD

**Discussion by ssivaguru** [ImageMagick/ImageMagick#2252] [Help] [answered]
```
    #include <stdio.h>
    #include <stdlib.h>
    #include <string.h>
    #include <time.h>
    #include <MagickCore/MagickCore.h>

    int main(int argc, char **argv)
    {
        ExceptionInfo
            *exception;

        Image
            *Resized_image,
            *images;

        ImageInfo
            *image_info;

        /*
        Initialize the image info structure and read an image.
    */
        MagickCoreGenesis(*argv, MagickTrue);

        MagickInfo *entry;

        exception = AcquireExceptionInfo();
        image_info = CloneImageInfo((ImageInfo *)NULL);
        image_info->size = "1920x1080";
        image_info->depth = 8;
        /* this one works */
        (void)strcpy(image_info->filename, "image.rgba");
        images = ReadImage(image_info, exception);

        /* this one doesn't
        size_t length = 1920 * 1080 * 4;
        char *buffer = calloc(length, sizeof(char *));
        FILE *file = fopen("image.rgba", "r");
        fread(buffer, length, sizeof(char), file);
        images = BlobToImage(image_info, buffer, length, exception);
        */

        if (exception->severity != UndefinedException)
        {
            printf("%s\n", exception->reason);
            exit(1);
        }

        if (images == (Image *)NULL)
            exit(1);

        (void)strcpy(images->filename, "output.jpeg");
        WriteImage(image_info, images, exception);
        return (0);
    }
```

I am trying to read an RGBA image when I read it using "ReadImage" it seems to be working properly. but when I try to manually read the file and convert it into an Image its not working.
I am assuming that "BlobToImage" this is the method that we use to read the in-memory data if I am wrong do suggest the method which I am supported to use?
or if it's the correct method could someone tell me where I am going wrong (should i add more data to image info?)?

**Comment by snibgo** [comment, marked as answer]
The answer to your final query is "yes".

In your "this one works", IM gets the image format from the extension, "rgba", so it knows how to interpret the numbers in the file.

In your "this one doesn't", you never tell IM the format of the blob, so IM doesn't know how to interpret the data. Is it JPEG, or PNG, or raw RGBA? We know the data is RGBA, but you need to tell IM, before calling BlobToImage():
```
CopyMagickString(image_info->magick,"RGBA",MagickPathExtent);
```
EDIT to add: Set it back to the empty string before writing the output, or you will get a RGBA format file named "output.jpeg" which will be confusing.
```
CopyMagickString(image_info->magick,"",MagickPathExtent);
```

**Comment by ssivaguru** [comment]
ty.. now it's working properly...
